Compiler/src/Error.hs

```diff
@@ -23,15 +54,22 @@ instance PrettyPrint Error where
     prettyPrint (LookupError (LookupErrorType s)) = ["LookupError: Type " ++ s ++ " is not defined"]
     prettyPrint (LookupError (LookupErrorTypeDecl (TransformerDecl tx _ _ _))) = ["LookupError: expected type but got transformer" ++ show tx]
 
--- dummy values that are returned as proxies when errors are encountered
-dummyBaseType :: BaseType
-dummyBaseType = Bot
-
-dummyType :: Type
-dummyType = (Any, Bot)
+groupErrors :: [ErrorCat] -> ([Error Preprocessing], [Error Typechecking], [Error Compiling])
+groupErrors [] = ([], [], [])
+groupErrors (x:xs) = (groupParsed ++ packedParsed, groupPreprocessed ++ packedPreprocessed, groupTypechecked ++ packedTypechecked)
+    where
+        (packedParsed, packedPreprocessed, packedTypechecked) = case x of
+            (PreprocessorError e) -> ([e], [], [])
+            (TypecheckerError e) -> ([], [e], [])
+            (CompilerError e) -> ([], [], [e])
```

Comment:
The default `Monoid` instances for tuples should help you out here, so keep something like this as a function `groupError` or something, then `groupErrors` becomes `groupErrors = mconcat . map groupError`